### PR TITLE
Added OrderInstructions message

### PR DESCRIPTION
**Pull request by thehenrytsai**

### CODE REVIEW

**Review by ethan-tbd — APPROVED** [review]

**Comment by jiyoontbd** [comment]
@thehenrytsai could you remove payment instructions from quote.data.payin and quote.data.payout?

https://github.com/TBD54566975/tbdex/tree/main/specs/protocol#quote

image

File: packages/protocol/src/message-kinds/order.ts [hunk 1]
```diff
@@ -16,12 +16,14 @@ export type CreateOrderOptions = {
  */
 export class Order extends Message {
   /** a set of valid Message kinds that can come after an order */
-  readonly validNext = new Set<MessageKind>(['orderstatus'])
+  readonly validNext = new Set<MessageKind>(['orderinstructions'])
```
Comment (jiyoontbd) [inline packages/protocol/src/message-kinds/order.ts:19]: i have these in tbdex-go

```go
// ValidNext returns the valid message kinds that can follow an order.
func ValidNext() []string {
	return []string{orderstatus.Kind, closemsg.Kind, cancel.Kind}
}
```

File: packages/protocol/tests/exchange.spec.ts [hunk 2]
```diff
@@ -258,20 +273,43 @@ describe('Exchange', () => {
...
+      it('can add an OrderInstructions after Order', async () => {
         const exchange = new Exchange()
 
         exchange.addMessages([rfq, quote, order])
 
+        exchange.addNextMessage(orderInstructions)
+        expect(exchange.orderInstructions).to.deep.eq(orderInstructions)
+      })
+
+      it('cannot add Rfq, Quote, Order, OrderStatus, or Close after Order', async () => {
```
Comment (jiyoontbd) [inline packages/protocol/tests/exchange.spec.ts:285]: technically you should be able to add a close after an order

**Review by jiyoontbd — APPROVED** [review]
couple comments to address, but otherwise looks good! thanks henry!

File: packages/protocol/tests/exchange.spec.ts (hunk 2 above)
Comment (thehenrytsai) [reply packages/protocol/tests/exchange.spec.ts:285]: Thanks, will update!

File: packages/protocol/src/message-kinds/order.ts (hunk 1 above)
Comment (thehenrytsai) [reply packages/protocol/src/message-kinds/order.ts:19]: Addressed.

File: packages/protocol/tests/exchange.spec.ts (hunk 2 above)
Comment (thehenrytsai) [reply packages/protocol/tests/exchange.spec.ts:285]: Addressed.

**Pull request merged**